File: VB/MessengerExample/Model/EmployeeContext.vb
Imports System.Data.Entity

Namespace MessengerExample.Model
    Public Class EmployeeContext
        Inherits DbContext

        Public Property Employees() As DbSet(Of Employee)
        Public Function IsNew(ByVal employee As Employee) As Boolean
            If employee Is Nothing Then
                Return False
            End If
            Dim state As EntityState = Entry(employee).State
            Return state = EntityState.Added
        End Function
        Public Function NeedSave(ByVal employee As Employee) As Boolean
            If employee Is Nothing Then
                Return False
            End If
            Dim state As EntityState = Entry(employee).State
            Return state = EntityState.Modified OrElse state = EntityState.Added
        End Function
    End Class
End Namespace

File: CS/MessengerExample/Model/EmployeeContext.cs
using System.Data.Entity;

namespace MessengerExample.Model {
    public class EmployeeContext : DbContext {
        public DbSet<Employee> Employees { get; set; }
        public bool IsNew(Employee employee) {
            if(employee == null) return false;
            EntityState state = Entry(employee).State;
            return state == EntityState.Added;
        }
        public bool NeedSave(Employee employee) {
            if(employee == null) return false;
            EntityState state = Entry(employee).State;
            return state == EntityState.Modified || state == EntityState.Added;
        }
    }
}
